### ISSUE TITLE
Psort.py elastic - Content-Type header [] is not supported

### ISSUE THREAD

**Issue by SekDev** [log2timeline/plaso#1827]
Latest version of Elastic Search v6.2.3 and latest Plaso (psort.py)

Using the python module "elasticsearch", trying to import a .plaso file into Elastic but failing with the following error.

(406, u'Content-Type header [] is not supported')

-----------------------------------------------------------------------------------------------------------
$ pip freeze | grep -i elastic             (show installed python elastic module, this is the latest)
elasticsearch==6.2.0

-----------------------------------------------------------------------------------------------------------

$ psort.py -V
plaso - psort version 20180407  (I also tried stable release plaso-20180127 - same error)

-------------------------------------------------------------------------------------------------------

$ psort.py -o elastic --server localhost --port 9200 --index_name dump --raw_fields image.plaso

2018-04-18 11:08:10,580 [INFO] (MainProcess) PID:22036 <elastic> Flush interval: 1000
2018-04-18 11:08:10,580 [INFO] (MainProcess) PID:22036 <elastic> Add non analyzed string fields: True
2018-04-18 11:08:10,580 [INFO] (MainProcess) PID:22036 <elastic> Elastic user: None
2018-04-18 11:08:10,580 [INFO] (MainProcess) PID:22036 <elastic> Elastic password: ****
2018-04-18 11:08:10,580 [WARNING] (MainProcess) PID:22036 <psort_tool> Appending to an already existing storage file.
Traceback (most recent call last):
  File "/usr/local/bin/psort.py", line 4, in <module>
    __import__('pkg_resources').run_script('plaso==20180407', 'psort.py')
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 719, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1504, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/plaso-20180407-py2.7.egg/EGG-INFO/scripts/psort.py", line 68, in <module>
    if not Main():
  File "/usr/local/lib/python2.7/dist-packages/plaso-20180407-py2.7.egg/EGG-INFO/scripts/psort.py", line 54, in Main
    tool.ProcessStorage()
  File "/usr/local/lib/python2.7/dist-packages/plaso-20180407-py2.7.egg/plaso/cli/psort_tool.py", line 561, in ProcessStorage
    time_slice=self._time_slice, use_time_slicer=self._use_time_slicer)
  File "/usr/local/lib/python2.7/dist-packages/plaso-20180407-py2.7.egg/plaso/multi_processing/psort.py", line 1001, in ExportEvents
    output_module.WriteHeader()
  File "/usr/local/lib/python2.7/dist-packages/plaso-20180407-py2.7.egg/plaso/output/elastic.py", line 327, in WriteHeader
    elastic_user=self._elastic_user)
  File "/usr/local/lib/python2.7/dist-packages/plaso-20180407-py2.7.egg/plaso/output/elastic.py", line 58, in __init__
    self._index = self._EnsureIndexExists(index_name, mapping)
  File "/usr/local/lib/python2.7/dist-packages/plaso-20180407-py2.7.egg/plaso/output/elastic.py", line 96, in _EnsureIndexExists
    index=index_name, body={'mappings': mapping})
  File "build/bdist.linux-x86_64/egg/elasticsearch/client/utils.py", line 69, in _wrapped
  File "build/bdist.linux-x86_64/egg/elasticsearch/client/indices.py", line 103, in create
  File "build/bdist.linux-x86_64/egg/elasticsearch/transport.py", line 307, in perform_request
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_urllib3.py", line 93, in perform_request
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/base.py", line 105, in _raise_error
elasticsearch.exceptions.TransportError: TransportError(406, u'Content-Type header [] is not supported')

--------------------------------------------------------------------------------------------------------------------
Elastic Version (http://localhost:9200/)

number | "6.2.3"
build_date | "2018-03-13 T10:06:29.741383Z"
lucene_version | "7.2.1"
minimum_wire_compatibility_version | "5.6.0"
minimum_index_compatibility_version | "5.0.0"

----------------------------------------------------------------------------------------------------------------------

Any ideas ? I originally tried with the elasticsearch-py module but it seemed very old and im fairly sure im using the correct latest Elastic python module now. Still no joy.

**Comment by SekDev** [comment]
I did a little more digging as I was getting mixed results depending on which elasticsearch python module version I was using. So I decided to remove the elasticsearch==6.2.0 and try elasticsearch==6.0.0 as some other people had suggested this might work. That did not work either, I was getting different errors.

Eventually I decided to try building the very latest official python elasticsearch module instead of using pip install and now psort.py seems to at least attempt to run but it fails to process any data. So maybe a little closer but still no cigar ?

https://www.elastic.co/guide/en/elasticsearch/client/python-api/current/index.html
https://github.com/elastic/elasticsearch-py

Latest stable release was 6.2.0 same as pip, so I grabbed the nightly via git clone.

$ git clone https://github.com/elastic/elasticsearch-py.git
$ cd elasticsearch-py
$ sudo python setup.py install

The version of the nightly build is still 6.2.0 which is super confusing, but anyway ...

$ python
>>> import elasticsearch
>>> print elasticsearch.__version__
(6, 2, 0)

Try psort again now,

$ psort.py -o elastic --server localhost --port 9200 --index_name dump --raw_fields image.plaso

plaso - psort version 20180407

Storage file	: ../image.plaso

Identifier              PID     Status          Memory          Events          Tags            Reports
Main                    26236   idle            97.9 MiB        0 (0)           0 (0)           0 (0)

2018-04-20 10:13:35,965 [WARNING] (MainProcess) PID:26236 <elastic> Empty value passed for a required argument 'body'.
2018-04-20 10:13:35,966 [INFO] (MainProcess) PID:26236 <elastic> 0 events added
Processing completed.

******************************** Export results ********************************
       Events filtered : 0
Events from time slice : 0
      Events processed : 0
--------------------------------------------------------------------------------

My experience is that there are multiple versions of the elasticsearch module floating around and its rather confusing for the community. If the best version can be determined that would be great.

Also, I don't know why "Empty value passed for a required argument 'body'." is causing psort.py to still fail now at the importing events stage.

**Comment by joachimmetz** [comment]
blocked by #1828

**Comment by joachimmetz** [comment]
We currently support elasticsearch >=5.4.0,< 6.0.0 (or elasticsearch-py) and not pyelasticsearch

So I assume this is a duplicate of https://github.com/log2timeline/plaso/issues/1653
